### PR TITLE
nix.conf: move write for 'fallback', always set it

### PR DESCRIPTION
**Pull request by colemickens**

### CODE REVIEW

File: magic-nix-cache/src/main.rs [hunk 1]
```diff
@@ -222,6 +222,11 @@ async fn main_cli() -> Result<()> {
         .open(&nix_conf_path)
         .with_context(|| "Creating nix.conf")?;
 
+    // always enable fallback, first
+    nix_conf
+        .write_all(format!("fallback = true\n",).as_bytes())
```
Comment (cole-h) [inline magic-nix-cache/src/main.rs:227]:
```suggestion
        .write_all(b"fallback = true\n")
```

Comment (colemickens) [reply magic-nix-cache/src/main.rs]: fixed; thanks

File: magic-nix-cache/src/main.rs [hunk 2]
```diff
@@ -222,6 +222,11 @@ async fn main_cli() -> Result<()> {
         .open(&nix_conf_path)
         .with_context(|| "Creating nix.conf")?;
 
+    // always enable fallback, first
+    nix_conf
+        .write_all(b"fallback = true\n")
+        .with_context(|| "Updating with nix.conf with fallback")?;
```
Comment (cole-h) [inline magic-nix-cache/src/main.rs:228]:
```suggestion
        .with_context(|| "Setting fallback in nix.conf")?;
```

Comment (colemickens) [reply magic-nix-cache/src/main.rs]: sure, fixed

**Review by cole-h — APPROVED** [review]

**Pull request merged**